### PR TITLE
chore!: remove make-dir package and use native, drop Node.js 10 support

### PR DESCRIPTION
**Pull request by XhmikosR**
This requires a change in the engines version since `fs.mkdir`'s recursive option requires >= 10.12.0.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "fetch"
   ],
   "engines": {
-    "node": ">=10"
+    "node": ">=12"
```
Comment (erezrokah) [inline package.json:36]:
```suggestion
    "node": "^12.20.0 || ^14.14.0 || >=16.0.0"
```

To match https://github.com/netlify/cli/blob/f560c82885df081c94b53ae2e3db1b6e4bf839da/package.json#L14

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @XhmikosR, added a comment.

Also, to force a breaking change and a major version bump, you can use the following commit message:

`chore!: drop support for Node.js <= 10`.

See example https://github.com/netlify/cli/pull/3648

File: package.json (hunk 1 above)
Comment (XhmikosR) [reply package.json:36]: I thought about it too, but this package seems to work with Node.js >= 12.

Feel free to change it if you think it's worth it, I tend to limit engines based on what's really needed :)

Comment (XhmikosR) [reply package.json:39]: Oh, BTW we should also change the TS Config file. What's the ES equivalent for Node.js 12.x?

Comment (XhmikosR) [reply package.json]: All right, I think this should be OK now. I used `ES2019` and your engines suggestion, although arguably Node.js >=12 works too in this project.

Comment (erezrokah) [reply package.json]: I believe those are the minimal versions required to support ES Modules without an additional flag.
At some point we'd want to migrate this lib to ES modules

Comment (XhmikosR) [reply package.json]: Yeah, that's true.

**Pull request merged**